### ISSUE TITLE
The results on the WSP model are not consistent with those in the paper, and there is a big difference, and what is wrong？

### ISSUE THREAD

**Issue by foolishcqx** [Yuto-Matsunaga/Prompt_Tuning_for_Audio_Deepfake_Detection#2]
The results on the WSP model are inconsistent, and the results obtained by different random seeds are quite different and extremely unstable
![微信图片_20241028215959](https://github.com/user-attachments/assets/5ae0a7cc-5beb-4686-a673-ed676fe98da8)

**Comment by foolishcqx** [comment]
It's worth mentioning that I just modified the dataset to the naming method required by the code, and didn't make any other changes, and the dataset division was also used in the code！ Would be grateful if you answered my question！

**Comment by Yuto-Matsunaga** [comment]
It is possible that the training was performed with the default values for the hyper-parameters in uniform_config.yaml. Please refer to the values in the CSV file in the results directory and adjust the target parameters according to your specific case.
The values in the CSV file are as follows, from left to right:
1.  Number of tokens
2.  Number of training samples
3.  Batch size
4.  Hyper-parameters for CBloss
5.  Learning rate
6.  Weight decay
I apologize for not including this information in the README.

**Comment by foolishcqx** [comment]
Thanks for the reply, I've tried to change the parameters to the values in the csv file to test, but the results are similar to the above, still very unstable, is there anything else to pay attention to? Thank you very much for your answer！

**Comment by Yuto-Matsunaga** [comment]
Thank you for your reply.
First of all, could you provide more information about the exact commands you used to get the result?

**Comment by foolishcqx** [comment]
Sorry, this is the main information, I just modified the code in the following part to make it easier to accommodate the parameters of different datasets。The content of A_uniform_config.yaml as below：
![微信截图_20241030170634](https://github.com/user-attachments/assets/4356a682-06f9-499e-8038-97c6391d2964)
```
if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument('--device', type=str, default='cuda:0')
    parser.add_argument('--dataset_name', type=str, default='LA', choices=['HABLA', 'VCC_ENG', 'VCC_FIN', 'VCC_MAN', 'VCC_GER', 'In_The_Wild','LA'])
    parser.add_argument('--seed', type=int, default=42)
    parser.add_argument('--n_tokens', type=int, default=None)
    parser.add_argument('--n_samples', type=int, default=None)
    parser.add_argument('--target_params', type=str, nargs="*", default=None)
    parser.add_argument('--beta', type=float, default=None, help='for Class Balanced loss')
    main_args = parser.parse_args()
    dir_path =  'A'
    # load uniform config params
    with open('A_uniform_config.yaml', 'r') as f:
        config = yaml.safe_load(f)
    for data_set in ['VCC_ENG', 'VCC_FIN', 'VCC_MAN', 'VCC_GER', 'HABLA', 'In_The_Wild','LA']:
        csv_dir = "./results"
        hyparamater_csv = os.listdir(csv_dir)
        for i in hyparamater_csv:
            if not i.endswith('.csv'):
                continue
            if not (dir_path+'.csv' in i and data_set in i):
                continue
            informations = pd.read_csv(os.path.join(csv_dir, i)).columns

            if len(informations) == 5:
                config['n_tokens'] = 5
                config['n_samples'], config['batch_size'], config['beta'], config['lr'], config['weight_decay'] = int(informations[0]),int(informations[1]),float(informations[2]),float(informations[3]),float(informations[4])
            else:
                config['n_tokens'], config['n_samples'], config['batch_size'], config['beta'], config['lr'], config['weight_decay'] = int(informations[0]),int(informations[1]),int(informations[2]),float(informations[3]),float(informations[4]), float(informations[5])

            print(i)
        # variable parameters priortize main_args
        config["device"] = main_args.device
        config["dataset_name"] = data_set
        if main_args.n_tokens is not None: config["n_tokens"] = main_args.n_tokens
        if main_args.n_samples is not None: config["n_samples"] = main_args.n_samples

        seed_list = [1,42,57,123,256,345,567,890,1024,2345,3478,9999]
        #seed_list = [1]
        eer_cm_list = []
        eem_result_path = os.path.join('results', dir_path)
        for seed in seed_list:
            #if main_args.seed != 42: config["seed"] = main_args.seed
            print(f'starting experiment with seed: {seed}')
            config["seed"] = seed
            if main_args.target_params is not None: config["target_params"] = main_args.target_params
            if main_args.beta is not None: config["beta"] = main_args.beta

            if config['dataset_name'] == 'HABLA':
                config["database_dir"] = "HABLA/asvspoof2019_LA-20241010T134714Z-001/asvspoof2019_LA" # change for /path/to/your_dataset
            elif config['dataset_name'] in ['VCC_ENG', 'VCC_FIN', 'VCC_MAN', 'VCC_GER']:
                config["database_dir"] = "VCC2020" # change for /path/to/your_dataset
            elif config["dataset_name"] == 'In_The_Wild':
                config["database_dir"] = "release_in_the_wild" # change for /path/to/your_dataset
            elif config["dataset_name"] == 'LA':
                config["database_dir"] = "ASVspoof2021_LA_eval"
            else:
                raise NotImplementedError

            # change for /path/to/your_dataset
            config["train_meta"] = f'LA-keys-stage-1/keys/CM/train_metadata_{config["dataset_name"]}.txt'
            config["dev_meta"] = f'LA-keys-stage-1/keys/CM/dev_metadata_{config["dataset_name"]}.txt'
            config["test_meta"] = f'LA-keys-stage-1/keys/CM/test_metadata_{config["dataset_name"]}.txt'
            if config["dataset_name"] == 'LA':
                config["train_meta"] = f'LA-keys-stage-1/keys/CM/trial_metadata.txt'
                config["dev_meta"] = f'LA-keys-stage-1/keys/CM/trial_metadata.txt'
                config["test_meta"] = f'LA-keys-stage-1/keys/CM/trial_metadata.txt'

            # The experimental folder name is determined dynamically.
            # Automatically generates a sequential number of 3-digit numbers.
            if not os.path.exists(os.path.join('results',dir_path, config["dataset_name"] + "_seed_" + str(config["seed"]))):
                os.makedirs(os.path.join('results', dir_path, config["dataset_name"] + "_seed_" + str(config["seed"])))
            config["exp_name"] = make_foldername_by_seq(os.path.join('results', dir_path, config["dataset_name"] + "_seed_" + str(config["seed"])), prefix='exp', seq_digit=3)
            result_path = os.path.join('results', dir_path, config["dataset_name"] + "_seed_" + str(config["seed"]), config["exp_name"])
            config["result_path"] = result_path
            print(f'result_path: {result_path}')    
            if not os.path.exists(result_path):
                os.makedirs(result_path)
            else:
                raise FileExistsError
            
            # fix seed
            random.seed(config["seed"])
            torch.manual_seed(config["seed"])
            torch.backends.cudnn.benchmark = False
            torch.backends.cudnn.deterministic = True
            g = torch.Generator()
            g.manual_seed(config["seed"])

            # good luck!!
            if config["eval_only"]:
                model = load_model(config, config["device"])
                eer_cm = eval(config, model)
            else:
                eer_cm = main(config)
            eer_cm_list.append(eer_cm)
            print(f'EER_CM: {eer_cm}')
            with open(os.path.join(eem_result_path, config["dataset_name"] + '_eer_cm.txt'), 'a', encoding='utf-8') as f:
                f.write("seed =:{}, EER_CM =:{}\n".format(config["seed"], str(eer_cm)))
        with open(os.path.join(eem_result_path, config["dataset_name"] + '_eer_cm.txt'), 'a', encoding='utf-8') as f:
            f.write("Average EER_CM =:{}, Stadard Deviation =:{}\n".format(sum(eer_cm_list)/len(eer_cm_list), np.std(eer_cm_list)))
```
Here are the results for each dataset
[VCC_MAN_eer_cm.txt](https://github.com/user-attachments/files/17568844/VCC_MAN_eer_cm.txt)
[VCC_FIN_eer_cm.txt](https://github.com/user-attachments/files/17568845/VCC_FIN_eer_cm.txt)
[VCC_ENG_eer_cm.txt](https://github.com/user-attachments/files/17568846/VCC_ENG_eer_cm.txt)
[LA_eer_cm.txt](https://github.com/user-attachments/files/17568847/LA_eer_cm.txt)
[In_The_Wild_eer_cm.txt](https://github.com/user-attachments/files/17568848/In_The_Wild_eer_cm.txt)
[HABLA_eer_cm.txt](https://github.com/user-attachments/files/17568849/HABLA_eer_cm.txt)
[VCC_GER_eer_cm.txt](https://github.com/user-attachments/files/17568850/VCC_GER_eer_cm.txt)

**Comment by Yuto-Matsunaga** [comment]
Sorry for the delayed response. Upon checking "prompt_tuning.py", I noticed that the argument "eval_set= Dataset_xx" had "n_samples=config["n_samples"]" set. This seems to have inadvertently reduced the number of test data samples. Please try removing the relevant part and test it again.
